src/Microsoft.Health.Fhir.Core/Features/Operations/DataConvert/DataConvertEngine.cs

```diff
@@ -0,0 +1,150 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.Extensions.Logging;
+using Microsoft.Health.Fhir.Core.Exceptions;
+using Microsoft.Health.Fhir.Core.Features.Operations.DataConvert.Models;
+using Microsoft.Health.Fhir.Core.Messages.DataConvert;
+using Microsoft.Health.Fhir.Liquid.Converter;
+using Microsoft.Health.Fhir.Liquid.Converter.Exceptions;
+using Microsoft.Health.Fhir.Liquid.Converter.Hl7v2;
+using Microsoft.Health.Fhir.TemplateManagement;
+using Microsoft.Health.Fhir.TemplateManager.Exceptions;
+
+namespace Microsoft.Health.Fhir.Core.Features.Operations.DataConvert
+{
+    public class DataConvertEngine : IDataConvertEngine
+    {
+        private readonly IContainerRegistryTokenProvider _containerRegistryTokenProvider;
+        private readonly ITemplateProviderFactory _templateProviderFactory;
+        private readonly ILogger<DataConvertEngine> _logger;
+
+        private readonly Dictionary<string, IFhirConverter> _dataConverterMap = new Dictionary<string, IFhirConverter>();
+        private const char ImageRegistryDelimiter = '/';
+        private const string DefaultTemplateReference = "microsofthealth/fhirconverter:default";
+
+        public DataConvertEngine(
+            IContainerRegistryTokenProvider containerRegistryTokenProvider,
+            ITemplateProviderFactory templateProviderFactory,
+            ILogger<DataConvertEngine> logger)
+        {
+            EnsureArg.IsNotNull(containerRegistryTokenProvider, nameof(containerRegistryTokenProvider));
+            EnsureArg.IsNotNull(templateProviderFactory, nameof(templateProviderFactory));
+            EnsureArg.IsNotNull(logger, nameof(logger));
+
+            _containerRegistryTokenProvider = containerRegistryTokenProvider;
+            _templateProviderFactory = templateProviderFactory;
+            _logger = logger;
+
+            _dataConverterMap.Add(DataConvertInputDataType.Hl7v2.ToString(), new Hl7v2Processor());
```

Comment:
Any reason for converting the enum to string vs. just using the enum as the key?

Reply:
Should be better to use the enum here. Thanks!